wagtailio/navigation/models.py

```diff
@@ -0,0 +1,126 @@
+from django.db import models
+
+from modelcluster.fields import ParentalKey
+from modelcluster.models import ClusterableModel
+from wagtail.admin.edit_handlers import (
+    FieldPanel,
+    InlinePanel,
+    MultiFieldPanel,
+    StreamFieldPanel,
+)
+from wagtail.contrib.settings.models import BaseSetting, register_setting
+from wagtail.core.fields import StreamField
+from wagtail.core.models import Orderable
+from wagtail.snippets.edit_handlers import SnippetChooserPanel
+from wagtail.snippets.models import register_snippet
+
+from wagtailio.navigation.blocks import NavItemBlock, NavItemCTA, NavStreamField
+from wagtailio.navigation.forms import MainMenuForm
+
+
+@register_snippet
+class FooterMenu(models.Model):
+    name = models.CharField(max_length=255)
+    sections = StreamField(
+        NavStreamField(),
+    )
+
+    panels = [
+        FieldPanel("name"),
+        StreamFieldPanel("sections"),
+    ]
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        verbose_name = "Footer menu"
+
+
+@register_snippet
```

Comment:
@zerolab I'm happy to flesh this out with you, personally, I don't think navigation UI is that well suited here for snippets. It works but I think the workflow is awkward.

To set this we'd need to in this order; create Footer, Main menu and Get started items and then add them to the main add them to navigation settings. I think that it might be more appropriate to do something more akin to a menu settings hook using the clusterable model (so we can create different menus for the sites? not sure if that's needed) and then adding the streamfields there in one place.

What do you think?

Reply:
I am not very keen on one massive Settings instance with large StreamFields.
The idea behind using snippets is for content staging and for previews, should we need to.

`MainMenu` could potentially be simplified to be one big StreamField, though, as I agree the workflow can be improved. Basically `MainMenuSection` becoming a StreamField block in `MainMenu`, rather than FK

Reply:
I've updated this one in 1f6ef38f